### DISCUSSION TITLE
Unable to recover passwords in firefox 94 after ubuntu 21.10 migration

### DISCUSSION THREAD

**Discussion by HDInfautre** [keepassxreboot/keepassxc#7174] [Q&A] [answered]
Hello

I have several kdbx with KeePassXC - Version 2.6.6 - Revision: 9c108b9 - Distribution: AppImage
I have several firefox profiles. 
I have linked each firefox profile with a kdbx. It worked fine!

I made the migration to ubuntu 21.10 with firefox 94.0.2 (snap)
I did not make any modification in KeePassXC and the connection to the browsers is activated.
now it fails all the time
I have the following messages
"Cannot connect to KeePassXC. Check that browser integration is enabled in KeePassXC settings."
OR
"KeePassXC-Browser encountered an error:
Key exchange failed. "

Regards

**Comment by droidmonkey** [comment, marked as answer]
You can't connect using the browser extension if the browser is running as a snap. It is impossible.

**Comment by HDInfautre** [comment]
Hi
Ah I understood Ubuntu obliges the use of firefox snap. newbie I don't know how to do, but it's unbearable to copy/paste the login and password !

a) How to make keepassXC work with firefox. You have to delete snap and put deb? but the next ubuntu version will overwrite it? and I won't have the firefox update the next month?
b) would firefox appimage work better with keepassXC appimage?
c) You will never integrate keepassXC with firefox snap (patch) ?

**Reply by droidmonkey** [reply]
You'll need to use Auto-Type or clipboard. This isn't something we can fix, it is a snap problem and all native messaging based extensions will fail to work.

**Comment by HDInfautre** [comment]
ok i understand c) never because impossible with snap format  and a) difficult with copy/paste!
For all users like me ubuntu > 21.10 and firefox >94, what would work?
would there be another couple of possibilities?
firefox appImage & keeepassXC appImage ?

**Reply by droidmonkey** [reply]
Never used Firefox appimage but it should work.
